src/index.ts

```diff
@@ -222,35 +222,29 @@ function encodeNumber(buffer: Uint8Array, i: number, number: Numeric): number {
   return i
 }
 
-export class StringTable extends Array {
-  #encodings = new Map<string, Uint8Array>()
-  #positions = new Map<string, number>()
+export const emptyTableToken = Symbol()
 
-  static from(values: StringTable | Array<string>): StringTable {
-    if (values instanceof StringTable) {
-      return values
-    }
+export class StringTable {
+  strings = new Array<string>()
+  #encodings = new Array<Uint8Array>()
+  #positions = new Map<string, number>()
 
-    // Need to copy over manually to ensure the lookup map is correct
-    const table = new StringTable()
-    for (const value of values) {
-      table.#positions.set(value, table.push(value) - 1)
-      table.#encodings.set(value, StringTable._encodeString(value))
+  constructor(tok?: typeof emptyTableToken) {
```

Comment:
Why this empty table token thing rather than something like this?

```js
  constructor(initial = ['']) {
    if (Array.isArray(initial)) {
      for (const item of initial) {
        this.dedup(item)
      }
    }
  }
```

Reply:
What if `initial` does not contain `''` as first element ?
The first version if proposed was this:
```node
constructor(values?: Array<string>) {
    if (values === undefined) {
      // Create a StringTable and populate first elemnt with ''
      this.dedup('')
    } else {
      // Need to copy over manually to ensure the lookup map is correct
      // Should probably check that first element is the empty string
      for (const value of values) {
        this.#positions.set(value, this.strings.push(value) - 1)
        this.#encodings.set(value, StringTable._encodeString(value))
      }
    }
  }
```
To which you suggested:
> Maybe add something like this?
> 
> ``` 
> if (!values.includes('')) {
>   values = [''].concat(values)
> }
> ```

That's why I removed the ability to pass a string array, and therefore the need of having to check that this array is valid.
The empty token allows to explicitly request an empty table. It would be even better if it was not exported, since there would be no way for the module client to build an empty table, but I needed the token for the tests.

Reply:
They'd have to explicitly pass in an empty array. If they omit the argument altogether it will use the `['']` default. I guess the empty token thing is acceptable to me, just feels really awkward/unidiomatic. It's only intended for internal use though so... 🤷🏻